content/react-part-2/reading/js-in-react/index.md

```diff
@@ -0,0 +1,435 @@
+
+
+### Template literals
+
+You can use template literals to work with strings.  This can be easier than concatenation and more flexible than hard coding.  These can be very useful with loops, conditionals, and lists. 
+
+Template Literal syntax:
+- Encase the entire template literal within `{ }` since it is a JavaScript string 
+- Enclose the entire string in backticks `` 
+- Variables need to be inside curly braces `{variableNameHere}`
+- `$` are escape tokens and introduce each variable within the template literal `${variableNameHere}`
+
+Should look something like this when you put it all together: `{'Hello, ${variableNameHere}'}`
+
+{{% notice blue "Example: Template Literals" "rocket" %}} 
+```react{linenos=table,hl_lines=[],linenostart=1}
+let num1 = 4;
+let num2 = 7;
+num2 = 8
+
+function TemplateLiteral(){
+   return(
+      <div>
+         {`(${num1} * ${num2}) is ${num1 * num2} and (${num1} + ${num2}) is ${num1 + num2}`}
+      </div>
+   );
+};
+
+export default TemplateLiteral;
+```
+{{% /notice %}}
+
+{{% expand "Check the output" %}}
+
+   (4 * 8) is 32 and (4 + 8) is 12
+
+{{% /expand %}}
+
+
+
+### Arrow Functions
+**Arrow functions** are simplified functions. You can use an arrow function as an argument for other functions.  They do not affect the global scope of the project.  Due to their scope, they can be very useful for defining `props`, loops, and conditionals. These functions will return values without using the `return` keyword.
+
+Why not only use arrow functions?  For long functions.  They are great for short expressions but longer functions can be hard to read.
+
+{{% notice blue "Example: Comparison of Functions" "rocket" %}} 
+```react{linenos=table,hl_lines=[],linenostart=1}
+//comparing functions
+
+function Greeting(){
+  return( 
+      <h1>Hello!  I am a standard function.</h1>   
+   );
+}
+
+const ArrowGreeting = () => (
+      <h1>Hello! I am an arrow function.</h1>
+);
+```
+{{% /notice %}}
+
+{{% expand "Check the output" %}}
+
+## Hello! I am a standard function.
+
+## Hello! I am an arrow function.
+
+{{% /expand %}}
+
+{{% notice blue "Example: Combining Functions" "rocket" %}} 
+```react{linenos=table,hl_lines=[],linenostart=1}
+//function with an arrow function
+
+function ShinyButton(){
+   const handleClick = () => {
+      alert("You clicked me!");
+   }
+   return(
+      <button onClick={handleClick}>
+         Click This Button!
+      </button>
+   );
+}
+
+export default ShinyButton;
+```
+{{% /notice %}}
+
+{{% expand "Check the output" %}}
+
+![ShinyButton Output](./images/ShinyButtonOutput.png)
+
+{{% /expand %}}
+
+
+### Working with Props and JS
+//TODO: format as relref link
+
+Recall from the [PREVIOUS CHAPTER](react-lsn1/reading/components/#functional-components), `props` are arguments passed to and between functions.  We can use anything as a `props`, since `props` are placeholders.
```

Comment:
Suggestion not to capitalize PREVIOUS CHAPTER just use previous chapter.  And please fix the link it is not working.

Reply:
link fixed! woot!